theano/gpuarray/dnn.py

```diff
@@ -2732,6 +2732,191 @@ def local_abstractconv3d_cudnn_graph(op, context_name, inputs, outputs):
+
+        assert cudnn.cudnnDataType_t.has_alias(precision)
+        self.precision = precision
+
+    def make_node(self):
+        node = Apply(self, [],
+                     [CDataType("cudnnSpatialTransformerDescriptor_t",
+                                freefunc="cudnnDestroySpatialTransformerDescriptor")()])
+        # DebugMode cannot compare the values of CDataType variables, so by
+        # default it returns False all the time. To prevent DebugMode from
+        # complaining because of the MergeOptimizer, we make this variable
+        # always compare to True.
+        out = node.outputs[0]
+        out.tag.values_eq_approx = tensor.type.values_eq_approx_always_true
+        return node
+
+    # Number of images
+    nimages = property(lambda self: self.dimensions[0])
+    # Number of channels
+    nchannels = property(lambda self: self.dimensions[1])
+    # Grid height
+    height = property(lambda self: self.dimensions[2])
+    # Grid width
+    width = property(lambda self: self.dimensions[3])
+    # Number of dimensions in the output tensor
+    nb_dims = property(lambda self: len(self.dimensions))
+
+    def c_code_cache_version(self):
+        return (super(GpuDnnSpatialTfDesc, self).c_code_cache_version(), version())
+
+
+class GpuDnnGridGenerator(DnnBase):
+
+    """
+    This Op builds a spatial transformer grid generator for use in spatial transformer network
+    operations.
+    """
+
+    __props__ = ('precision',)
+    _cop_num_inputs = 3
+    _cop_num_outputs = 1
+
+    def __init__(self, precision):
+        DnnBase.__init__(self, ["c_code/spatialtf_grid.c"], "spatialtf_grid")
+
+        self.precision = precision
+
+    def dnn_context(self, node):
+        return node.outputs[0].type.context_name
+
+    def make_node(self, grid_dimensions, theta, desc):
+        context_name = infer_context_name(desc, theta)
+
+        grid_dimensions = as_tensor_variable(grid_dimensions)
+        theta = gpu_contiguous(as_gpuarray_variable(theta, context_name))
+
+        assert cudnn.cudnnDataType_t.has_alias(theta.dtype)
+        assert theta.ndim == 3
+
+        # Allocate GPU memory for grid of coordinates
+        grid = GpuArrayType(dtype=self.precision,
+                            broadcastable=(False, False, False, False,),
+                            context_name=context_name)()
+
+        return Apply(self, [grid_dimensions, theta, desc], [grid])
+
+    def L_op(self, inputs, outputs, output_grads):
+        pass
+
+
+class GpuDnnGridSampler(DnnBase):
+
+    """
+    This Op builds a spatial transformer grid sampler for use in spatial transformer network
+    operations.
+    """
+
+    __props__ = ('precision',)
+    _cop_num_inputs = 6
+    _cop_num_outputs = 1
+
+    def __init__(self, precision):
+        DnnBase.__init__(self, ["c_code/spatialtf_sampler.c"], "spatialtf_sampler")
+
+        self.precision = precision
+
+    def dnn_context(self, node):
+        return node.outputs[0].type.context_name
```

Comment:
As this `dnn_context()` has currently the same definition as in parent class `DnnBase`, you can remove it here.